src/cart/crt_group.c

```diff
@@ -1179,33 +1179,30 @@ crt_group_lookup(crt_group_id_t grp_id)
 	return (grp_priv == NULL) ? NULL : &grp_priv->gp_pub;
 }
 
+/* NB: no D_DEBUG or D_ERROR in this function, since this might
+ * be executed inside the D_DEBUG(), see d_vlog().
+ */
```

Comment:
You probably can't use D_ASSERT() or D_GOTO either then.

Reply:
We're likely to either deprecate D_GOTO or land this patch https://github.com/daos-stack/daos/pull/3907 so it would be better not to use it in critical sections of the code.